vispy/visuals/line/line.py

```diff
@@ -92,6 +92,12 @@ def __init__(self, pos=None, color=(0.5, 0.5, 0.5, 1), width=1,
 
         self._changed = {'pos': False, 'color': False, 'width': False,
                          'connect': False}
+        
+        self._bounds = None
+        if pos is not None:
```

Comment:
DRY: bounds should be calculated in only one place, and that should be inside the bounds method (and the result should be cached).